Tests/SwiftCliCoreTests/InputValidatorTest.swift

```diff
@@ -12,186 +12,349 @@ class ValidatorTests: XCTestCase {
+                Swift.Result<SwiftCliCore.Success, SwiftCliCore.InputError>.success(SwiftCliCore.Success.patternNestedArray([["p1", "p1"], ["p1", "p1", "p1"]])),
+                Swift.Result<SwiftCliCore.Success, SwiftCliCore.InputError>.success(SwiftCliCore.Success.patternNestedArray([["p1", "p1"], ["p1", "p1", "p1"]])),
+                Swift.Result<SwiftCliCore.Success, SwiftCliCore.InputError>.failure(SwiftCliCore.InputError.invalidRowWidth(invalidRowNumber: 2, expectedStitchCount: 2, actualCount: 3))
+            ]
+        )
+        expect(result).to(equal(expectedResult))
 
     }
 
-    func testMultipleIncorrectStitchesTypesShouldThrowMultipleErrors() throws {
-        let multipleIncorrectStitchesPattern = ["p1 p1", "g1 p1", "p1 g1", "g1 p1"]
-       let expectedErrors = InputError.multipleErrors(errors: [
-        InputError.invalidStitch(
-            invalidStitch: "g1",
-            rowLocation: Optional(2),
-            stitchIndexInRow: 1
-        ),
-        InputError.invalidStitch(
-            invalidStitch: "g1",
-            rowLocation: Optional(3),
-            stitchIndexInRow: Optional(2)
-        ),
-        InputError.invalidStitch(
-            invalidStitch: "g1",
-            rowLocation: Optional(4),
-            stitchIndexInRow: Optional(1)
-        )])
-        expect {
-            try self.inputValidator.validateInput(
-                pattern: multipleIncorrectStitchesPattern)
-        }
-        .to(throwError(expectedErrors))
+    func testMultipleIncorrectStitchesTypesAndEmptyRowShouldThrowMultipleErrors() throws {
+        let multipleIncorrectStitchesPattern = ["", "g1 p1", "p1 g1"]
+        let result = self.inputValidator.validateInput(
+            pattern: multipleIncorrectStitchesPattern,
+            knitFlat: false
+        )
+        let expectedResult = PatternDataAndPossibleErrors(
+            arrayOfStrings: ["", "g1 p1", "p1 g1"],
+            arrayOfArrays: [[], ["g1", "p1"], ["p1", "g1"]],
+            arrayOfRowInfo: [],
+            results: [
+                Swift.Result<SwiftCliCore.Success, SwiftCliCore.InputError>.failure(SwiftCliCore.InputError.emptyRow(row: Optional(1))),
+                Swift.Result<SwiftCliCore.Success, SwiftCliCore.InputError>.failure(
+                    SwiftCliCore.InputError.multipleErrors(
+                        errors: [
+                            SwiftCliCore.InputError.invalidStitch(invalidStitch: "g1", rowLocation: Optional(2), stitchIndexInRow: Optional(1)),
+                            SwiftCliCore.InputError.invalidStitch(invalidStitch: "g1", rowLocation: Optional(3), stitchIndexInRow: Optional(2))
+                        ]
+                    )
+                ),
+                Swift.Result<SwiftCliCore.Success, SwiftCliCore.InputError>.success(SwiftCliCore.Success.patternNestedArray([[], ["g1", "p1"], ["p1", "g1"]]))])
```

Comment:
I think you need to do `import Foundation` to not have to qualify `Result` and make the `Success` enum public so that you don't have to qualify it either.